config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -209,10 +229,27 @@ public JwtConfigurer jwkSetUri(String uri) {
 			return this;
 		}
 
+		public JwtConfigurer jwtConverter(Converter<Jwt, ? extends AbstractAuthenticationToken> jwtConverter) {
+			this.jwtConverter = jwtConverter;
+			return this;
+		}
+
 		public OAuth2ResourceServerConfigurer<H> and() {
 			return OAuth2ResourceServerConfigurer.this;
 		}
 
+		Converter<Jwt, ? extends AbstractAuthenticationToken> getJwtConverter() {
```

Comment:
It seems reasonable that boot would have a property like `spring.security.oauth2.resourceserver.scopeAttributeName`, in which case boot may publish an instance of a `Converter` as a bean for us to pick up.

How would we introduce something like that passively?

The reason I mention is because, while I agree that while your reasoning effectively works around the resolution issue I detailed above, it could be reintroduced later on, at which point we won't be able to change the type in the DSL to something more targetted.

Reply:
I think we can revisit that if/when the time comes. If we do need that support the converter can be automatically converted to a new interface type passively. So I'm not super concerned there.